force some flags always on in linkflags
--- a/SConstruct
+++ b/SConstruct
@@ -32,7 +32,7 @@
                   'IMdkit/IMValues.c']
 
 
-cflags='-O2 -g -pipe'
+cflags='-O2 -g -pipe '
 
 # options
 AddOption('--prefix', dest='prefix', type='string', nargs=1, action='store',
@@ -52,7 +52,6 @@
 
 env = Environment(ENV=os.environ,
                   CFLAGS=cflags, CXXFLAGS=cflags,
-                  LINKFLAGS=['-export-dynamic'],
                   CPPPATH=['.', 'IMdkit'])
 opts.Update(env)
 
@@ -62,11 +61,6 @@
 
 opts.Save('configure.conf', env)
 
-
-# set rpath
-if GetOption('rpath') is not None:
-    env.Append(LINKFLAGS='-Wl,-R -Wl,%s' % GetOption('rpath'))
-
 envvar = [('CC', 'CC'),
           ('CXX', 'CXX'),
           ('CFLAGS', 'CFLAGS'),
@@ -83,7 +77,11 @@
 
 env.Append(CFLAGS=extra_cflags)
 env.Append(CXXFLAGS=extra_cflags)
+env.Append(LINKFLAGS=['-export-dynamic'])
 
+# set rpath
+if GetOption('rpath') is not None:
+    env.Append(LINKFLAGS=['-Wl,-R', '-Wl,%s' % GetOption('rpath')])
 
 #
 #==============================configure================================
